### PR TITLE
Admin password reset

### PR DESCRIPTION
**Pull request by dampee**
There was an official but undocumented way to reset the password for the admin user

### CODE REVIEW

**Comment by callumbwhyte** [comment]
Note, this does not work if your site is in an upgrading state.

If you remove the version number from the web.config the migrations are checked against the DB - if any migrations are found to have not run Umbraco assumes an upgrade is in progress and requires (real) authentication.

**Comment by dampee** [comment]
Never tested this TBH. So I've added your comment in the PR.

**Comment by Shazwazza** [comment]
@callumbwhyte This does work and I did it the other day :) @dampee the issue is that there's a 2nd instruction which is to remove your connection string entirely. Then run the installer and specify the same connection string you had originally and all should be well, even in upgrade mode.

**Comment by callumbwhyte** [comment]
I did this on 7.15 the other day and it definitely didn't work! Didn't remove the connection string though... 😊

File: Reference/Security/reset-admin-password.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+While there are multiple blogposts and even a [package](https://our.umbraco.com/packages/developer-tools/umbraco-admin-reset/) to reset the password of the admin user. There is an [official way](https://twitter.com/Shazwazza/status/1141594930550206464) to reset credentials from your lost admin password.
+
+## Step one: clear the configuration status in the web.config
+
+This is done to trigger the installation wizard.
+
+That means that in your web.config file the appsettings should look like this:
+
+``` csharp
```
Comment (sofietoft) [inline Reference/Security/reset-admin-password.md:17]:
```suggestion
```csharp
```

File: Reference/Security/reset-admin-password.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+versionFrom: 8.0.0
+---
+
+# Reset admin password
+
+There is one default admin user in any Umbraco installation. This is the first user of the system.
+
+While there are multiple blogposts and even a [package](https://our.umbraco.com/packages/developer-tools/umbraco-admin-reset/) to reset the password of the admin user. There is an [official way](https://twitter.com/Shazwazza/status/1141594930550206464) to reset credentials from your lost admin password.
+
+## Step one: clear the configuration status in the web.config
```
Comment (sofietoft) [inline Reference/Security/reset-admin-password.md:11]:
```suggestion
## Step one: Clear the configuration status in the web.config
```

File: Reference/Security/reset-admin-password.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+This is done to trigger the installation wizard.
```
Comment (sofietoft) [inline Reference/Security/reset-admin-password.md:13]:
```suggestion
The first step is to clear the configuration status in the `web.config` file. This is done to trigger the installation wizard.
```

File: Reference/Security/reset-admin-password.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+``` csharp
+<appSettings>
+    <add key="Umbraco.Core.ConfigurationStatus" value="" />
+</appSettings>
+```
+
+## Step two: update the database
+
+Open de database and go to the `umbracoUser` database table.
```
Comment (sofietoft) [inline Reference/Security/reset-admin-password.md:25]:
```suggestion
Open the database and go to the `umbracoUser` database table.
```

File: Reference/Security/reset-admin-password.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+Search the user where the column id equals `-1`.
+
+Clear the field, and change the content of the `userpassword` field to `default`.
+
+## Step one: run the installer
```
Comment (sofietoft) [inline Reference/Security/reset-admin-password.md:30]:
```suggestion
## Step Three: Run the installer
```

File: Reference/Security/reset-admin-password.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+That means that in your web.config file the appsettings should look like this:
+
+``` csharp
+<appSettings>
+    <add key="Umbraco.Core.ConfigurationStatus" value="" />
+</appSettings>
+```
+
+## Step two: update the database
```
Comment (sofietoft) [inline Reference/Security/reset-admin-password.md:23]:
```suggestion
## Step two: Update the database
```

**Comment by sofietoft** [comment]
Thanks for adding the suggestions @dampee ! 😄 

Does that mean we're ready to merge this one? Not entirely sure where the discussion above landed.

**Comment by sofietoft** [comment]
Hi there!
I have an important message for you, regarding this PR:

Yesterday we tried to do some cleaning up and force-pushed to the master branch which now causes some problems with existing forks and local clones. 
We have fixed up all the open PRs so that they are still mergeable now.

With that said, if you've been working on a **fork of the repo** or in a **local clone**, those will not work anymore.
If you wish to push any new changes to this PR, you will need to delete your local clone or fork and create fresh ones.

If you have any local work in progress, then you copy it over from your old fork or clone, following these steps:

**Forks**
1. In the old clone of the old fork, switch to the branch you were working on
2. In the new clone of the new fork, create a new branch with the same name as the old one
3. Copy all the affected files from the old clone to the new clone
4. Repeat steps 1-3 if you've been working with multiple branches
5. You can now safely delete your old fork

**Clones**
1. In the old clone, switch to the branch you were working on
2. Create and checkout a new branch on the new clone, using the same name as the old clone
3. Copy all the affected files from the old clone to the new clone
4. Repeat steps 1-3 if you've been working with multiple branches
5. Delete the old clone of the docs repo

And with that, you should be all set once more.

I do apologize for the inconvenience, and please let me know if you have any questions!
/Sofie

**Pull request merged**